src/test/java/fakes/UserControllerTest.java

```diff
@@ -14,45 +16,137 @@
     @Test
     @org.junit.jupiter.api.Disabled
     void withValidInexistingUsername_returnsOK__NO_FAKE_DEMO() {
-      UserController ctrl = new UserController();
+      UserController ctrl = new UserController(new FakeUserValidator(true, true));
       User user = new User("kalua");
 
       Message result = ctrl.create(user);
 
-      Assertions.assertEquals(result.status, Message.Status.OK);
+      Assertions.assertEquals(result.status, Message.Status.NOT_OK);
     }
 
+
     @Test
     void withValidInexistingUsername_returnsOK__FAKE() {
       // TODO
       // 1. Test schneller machen
       // 2. UserController.create so beinflussen,
       //    dass einmal der "if"- und einmal der "else"-Fall durchlaufen wird
+      //Warum NullPointerException? Musste Mockdatabase erzeugen..
```

Comment:
Versuche das mit dem Debugger rauszufinden. Im UserController wird wohl irgendwo erwartet dass this.database gesetzt ist.

Reply:
Habe zwei UserController konstruktoren. Habe den falschen aufgerufen ( ohne this.database) mit dem zweiten hats funktioniert, da ich dort this.userVaildator, this.database gesetzt habe.